### PR TITLE
HVG-254 Feature/details your info

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoFragment.kt [hunk 1]
```diff
@@ -31,44 +31,57 @@ class YourInfoFragment : Fragment(R.layout.contract_detail_your_info_fragment) {
                     (adapter as? YourInfoAdapter)?.submitList(
                         homeSection(
                             it.address.fragments.addressFragment,
-                            it.squareMeters,
-                            it.saType.displayName(requireContext())
-                        ) + coinsuredSection(it.numberCoInsured)
+                            it.saType.displayName(requireContext()),
+                            it.squareMeters
+                        ) + coinsuredSection(it.numberCoInsured) +
+                            YourInfoModel.Paragraph(
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoFragment.kt:37]: This paragraph appears to be common to the `coinsuredSection`, could we add it to that function? That would reduce the repetition somewhat.

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoFragment.kt [hunk 2]
```diff
@@ -31,44 +31,57 @@ class YourInfoFragment : Fragment(R.layout.contract_detail_your_info_fragment) {
...
+                                getString(R.string.insurance_details_view_your_info_co_insured_footer)
+                            ) + YourInfoModel.Header(
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoFragment.kt:39]: This header, paragraph and button appears to be a logical 'section', perhaps we could extract it into a function as well?

File: app/src/main/res/values/button_themes.xml [hunk 3]
```diff
@@ -11,17 +11,22 @@
         <item name="backgroundTint">@color/hedvig_large_button_state</item>
         <item name="android:padding">@dimen/base_margin_double</item>
         <item name="shapeAppearance">?shapeAppearanceLargeComponent</item>
+        <item name="cornerRadius">@dimen/large_button_corner_radius</item>
```
Comment (hedvigoscar) [inline app/src/main/res/values/button_themes.xml:14]:
```suggestion
```

This corner radius should be controlled by the `shapeAppearance` instead.

File: app/src/main/res/values/button_themes.xml [hunk 4]
```diff
@@ -11,17 +11,22 @@
...
-    <style name="Hedvig.Button.Outlined" parent="Widget.MaterialComponents.Button.OutlinedButton" />
+    <style name="Hedvig.Button.Outlined" parent="Widget.MaterialComponents.Button.OutlinedButton" >
+        <item name="strokeColor">?colorPrimary</item>
+    </style>
 
     <style name="Hedvig.Button.Outlined.Large">
         <item name="android:padding">@dimen/base_margin_double</item>
         <item name="shapeAppearance">?shapeAppearanceLargeComponent</item>
+        <item name="strokeColor">?colorPrimary</item>
+        <item name="cornerRadius">@dimen/large_button_corner_radius</item>
```
Comment (hedvigoscar) [inline app/src/main/res/values/button_themes.xml:29]:
```suggestion
```

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoModel.kt [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+    data class Header(val text: String) : YourInfoModel()
 
     data class Row(
         val label: String,
         val content: String
     ) : YourInfoModel()
+
+    data class Paragraph(val text: String) : YourInfoModel()
+
+    object Button : YourInfoModel()
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoModel.kt:13]: I think this name might be a bit too generic. It's not any old `Button`, this button always has a specific text and action. Perhaps `OpenChat`? Or something like that.

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoAdapter.kt [hunk 6]
```diff
@@ -44,13 +52,7 @@ class YourInfoAdapter :
...
                     return invalid(data)
                 }
-
-                root.setText(
-                    when (data) {
-                        YourInfoModel.Header.Details -> com.hedvig.app.R.string.CONTRACT_DETAIL_HOME_TITLE
-                        YourInfoModel.Header.Coinsured -> com.hedvig.app.R.string.CONTRACT_DETAIL_COINSURED_TITLE
-                    }
-                )
+                root.text = data.text
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoAdapter.kt:55]: Why did we move away from this structure? 😬

**Review by hedvigoscar — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoAdapter.kt (hunk 6 above)
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/yourinfo/YourInfoAdapter.kt:55]: I wanted to add another header, didnt think much about it. Your imperfection might be cleaner ill go back and use what was before

**Review by hedvigoscar — APPROVED** [review]
Looks good!

**Pull request merged**